chore: upgrade cypress to v12

edoardo opened the pull request:
### Key features

1. update Cypress to latest version 12
2. adjust files and configuration

---

### Description

Upgrade Cypress dependencies and tests to run with latest Cypress version 12.

---

### TODO

-   [x] bump `@dhis2/cli-utils-cypress` when [this PR](https://github.com/dhis2/cli-utils-cypress/pull/393) is merged

---

### Known issues

-   [ ] some tests are failing, not expected

---

### Screenshots

<img width="1508" alt="Screenshot 2023-03-30 at 11 43 00" src="https://user-images.githubusercontent.com/150978/228796773-cea94a03-ce05-4b74-8876-d4a4e77b387d.png">

edoardo committed:
chore: update Cypress dependencies

edoardo committed:
fix: upgrade to Cypress 12

Mohammer5 committed:
chore: get cypress tests running

edoardo committed:
fix: add missing start-server-and-test dev dependency

edoardo committed:
test(cypress): fix URL matching in intercept

edoardo committed:
fix: add custom command for handling iframe content

edoardo committed:
fix: modified tests to work with iframe plugins

edoardo committed:
fix: disable Chrome security for cross origin iframes

edoardo committed:
test: re-enable some filter tests

jenniferarnesen commented on package.json:
```diff
@@ -43,25 +43,31 @@
-        "cy:open-stub": "cypress_dhis2_api_stub_mode=STUB d2-utils-cypress open --appStart 'yarn cypress:start'",
-        "cy:run-stub": "cypress_dhis2_api_stub_mode=STUB d2-utils-cypress run --tags '@nonmutating' --appStart 'yarn cypress:start' --record",
-        "cy:capture": "cypress_dhis2_api_stub_mode=CAPTURE yarn d2-utils-cypress run --appStart 'yarn cypress:start'"
+        "cypress:open:live": "yarn cypress open --env networkMode=live",
```
Since the network shim isn't in use, maybe we should simplify all this and just have  
`cy:run` and `cy.open` like with our other repos. There would also be changes in the support/e2e.js file for this.

jenniferarnesen committed:
chore: nightly failing to set up cypress (#2378)

The verify workflow should run on tags, so that a corresponding branch is created
in the d2-ci repository

Do not run the release step on PRs, but do allow it for pushes

yarn cypress install sometimes fails. Turns out it isn't necessary anyway
since the cypress-io gh action takes care of installing cypress.

dhis2-bot committed:
fix(translations): sync translations from transifex (dev)

Automatically merged.

janhenrikoverland committed:
docs: move docs to app (#2384)

dhis2-bot committed:
fix(translations): sync translations from transifex (dev)

Automatically merged.

dependabot[bot] committed:
chore(deps): bump json5 from 1.0.1 to 1.0.2 (#2200)

Bumps [json5](https://github.com/json5/json5) from 1.0.1 to 1.0.2.
- [Release notes](https://github.com/json5/json5/releases)
- [Changelog](https://github.com/json5/json5/blob/main/CHANGELOG.md)
- [Commits](https://github.com/json5/json5/compare/v1.0.1...v1.0.2)

---
updated-dependencies:
- dependency-name: json5
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jen Jones Arnesen <jennifer@dhis2.org>

dependabot[bot] committed:
chore(deps): bump @dhis2/analytics from 25.1.10 to 25.1.15 (#2399)

* chore: nightly failing to set up cypress (#2378)

The verify workflow should run on tags, so that a corresponding branch is created
in the d2-ci repository

Do not run the release step on PRs, but do allow it for pushes

yarn cypress install sometimes fails. Turns out it isn't necessary anyway
since the cypress-io gh action takes care of installing cypress.

* fix(translations): sync translations from transifex (dev)

Automatically merged.

* docs: move docs to app (#2384)

* chore(release): cut 100.0.1 [skip release]

## [100.0.1](https://github.com/dhis2/dashboard-app/compare/v100.0.0...v100.0.1) (2023-06-12)

### Bug Fixes

* **translations:** sync translations from transifex (dev) ([5b0b3b6](https://github.com/dhis2/dashboard-app/commit/5b0b3b64227229415cbd6efda23bdb2cf13e3305))

* chore(deps): bump @dhis2/analytics from 25.1.10 to 25.1.15

Bumps [@dhis2/analytics](https://github.com/dhis2/analytics) from 25.1.10 to 25.1.15.
- [Release notes](https://github.com/dhis2/analytics/releases)
- [Changelog](https://github.com/dhis2/analytics/blob/master/CHANGELOG.md)
- [Commits](https://github.com/dhis2/analytics/compare/v25.1.10...v25.1.15)

---
updated-dependencies:
- dependency-name: "@dhis2/analytics"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Jen Jones Arnesen <jennifer@dhis2.org>
Co-authored-by: @dhis2-bot <apps@dhis2.org>
Co-authored-by: Jan Henrik Øverland <janhenrik.overland@gmail.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dhis2-bot committed:
fix(translations): sync translations from transifex (dev)

Automatically merged.

dhis2-bot committed:
fix(translations): sync translations from transifex (dev)

Automatically merged.

jenniferarnesen committed:
chore: increase default cypress timeout to reduce test flakiness (#2456)

Also increase the EXTENDED_TIMEOUT

dependabot[bot] committed:
chore(deps): bump @dhis2/ui from 8.13.6 to 8.13.11 (#2462)

* chore: nightly failing to set up cypress (#2378)

The verify workflow should run on tags, so that a corresponding branch is created
in the d2-ci repository

Do not run the release step on PRs, but do allow it for pushes

yarn cypress install sometimes fails. Turns out it isn't necessary anyway
since the cypress-io gh action takes care of installing cypress.

* fix(translations): sync translations from transifex (dev)

Automatically merged.

* docs: move docs to app (#2384)

* chore(release): cut 100.0.1 [skip release]

## [100.0.1](https://github.com/dhis2/dashboard-app/compare/v100.0.0...v100.0.1) (2023-06-12)

### Bug Fixes

* **translations:** sync translations from transifex (dev) ([5b0b3b6](https://github.com/dhis2/dashboard-app/commit/5b0b3b64227229415cbd6efda23bdb2cf13e3305))

* chore(deps): bump @dhis2/ui from 8.13.6 to 8.13.11

Bumps [@dhis2/ui](https://github.com/dhis2/ui/tree/HEAD/collections/ui) from 8.13.6 to 8.13.11.
- [Release notes](https://github.com/dhis2/ui/releases)
- [Changelog](https://github.com/dhis2/ui/blob/master/CHANGELOG.md)
- [Commits](https://github.com/dhis2/ui/commits/v8.13.11/collections/ui)

---
updated-dependencies:
- dependency-name: "@dhis2/ui"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Jen Jones Arnesen <jennifer@dhis2.org>
Co-authored-by: @dhis2-bot <apps@dhis2.org>
Co-authored-by: Jan Henrik Øverland <janhenrik.overland@gmail.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jenniferarnesen committed:
chore: disable nightly scheduled test run (#2511)

This is temporary during summer 2023 when there is very little frontend or backend activity

edoardo committed:
chore: update cypress tooling dependencies

edoardo committed:
test: re-enable commented test

HendrikThePendric committed:
chore: align things with Cypress 12

HendrikThePendric commented:
I've taken the PR out of draft and added the "on hold" label, because the e2e tests don't run for draft PRs.

HendrikThePendric committed:
chore: empty commit to trigger e2e checks on ci

edoardo commented:
Superseded by https://github.com/dhis2/dashboard-app/pull/3040
